src/ds/openapi.h

```diff
@@ -49,16 +49,24 @@ namespace ds
       }
     }
 
-    static inline std::string remove_invalid_chars(const std::string_view& s_)
+    static inline std::string sanitise_components_key(
+      const std::string_view& s_)
     {
+      // From the OpenAPI spec:
+      // All the fixed fields declared above are objects that MUST use keys that
+      // match the regular expression: ^[a-zA-Z0-9\.\-_]+$."
       std::string s(s_);
 
       for (auto& c : s)
       {
-        if (c == ':')
+        if (
```

Comment:
I didn't think I'd ever say that, but I wonder if a std::regexp would be more readable here.